src/views/AddItem.jsx

```diff
@@ -12,17 +12,67 @@ export function AddItem({ listToken }) {
 		addItemToList();
 	};
 
+	// Removes excess spaces or special characters, and makes all leters lowercase.
+	const normalizeText = (item) => {
```

Comment:
Nice job on this normalize function! Really, really useful.

You could actually move this whole function outside of the React component (since it doesn't depend on any React state) so it doesn't have to get rebuilt on every render.

And, just to echo myself on slack, it's always a good idea when using RegEx to create a robust suite of unit tests to make sure it's doing exactly what we think its doing. Not something you need to do here, just looking towards the future.

Reply:
Thank you for this, I learned something new today! If the function doesn't depend on any react state it can go outside of the React component. ✅   

I'm looking forward to asking more about unit tests with RegEx during office hours 😁